### ISSUE TITLE
Can't create a linkage with more than four bars

### ISSUE THREAD

**Issue by victor-cortez** [HugoFara/pylinkage#2]
Hi. I want to create a linkage based on a four bar linkage but with extra linkages at the coupler, like a triangle pr a square. Those would not move. I thought about making the head triangle or square with Fixed joints but I get unsupoted operant type errors.

TypeError: unsupported operand type(s) for +: 'NoneType' and 'float'

How can I make a four bar linkage with extra unpowered links held by the coupler link?

I tried this code but the head still moves even tho the second crank should not move since the angle is 0

```
# Main motor
crank = pl.Crank(0, 1, joint0=(0, 0), angle=.31, distance=1, name="B")
# Close the loop
pin = pl.Pivot(3, 2, joint0=crank, joint1=(3, 0),
               distance0=3, distance1=1, name="C")

crank2 = pl.Crank(0, 2.5, joint0=crank, angle=0.0, distance=1.5, name="B2")
pin2 = pl.Pivot(3, 3, joint0=crank2, joint1=pin,
                distance0=3, distance1=1.5, name="C2")
fix = pl.Fixed(joint0=crank2, joint1=pin)
# Linkage definition
my_linkage = pl.Linkage(
    joints=(crank, pin, crank2, pin2),
    order=(crank, pin, crank2, pin2),
    name="My four-bar linkage"
)

# Visualization
pl.show_linkage(my_linkage)
```

**Comment by HugoFara** [comment]
Hi!

Fixed joints are better if you want to form elements in the same rigidbody, you can also overconstrain pivot joints, but it will cause slower computations and other issues, same as using static cranks. Could you send your original code with Fixed joints or a drawing of your desired linkage?

**Comment by victor-cortez** [comment]
<img width="497" alt="Untitled2" src="https://user-images.githubusercontent.com/7144657/192583243-8d8fe461-6061-4e05-b24a-22cbbd268378.png">
Those are the linkages I want to create.

My code with fixed joints was the following:
```
# Main motor
crank = pl.Crank(0, 1, joint0=(0, 0), angle=.31, distance=1, name="B")
# Close the loop
pin = pl.Pivot(3, 2, joint0=crank, joint1=(3, 0),
               distance0=3, distance1=1, name="C")
# The triangle on the couple link
fix1 = pl.Fixed(joint0=crank, joint1=(0, 2))
fix2 = pl.Fixed(joint0=fix1, joint1=pin)
# Linkage definition
my_linkage = pl.Linkage(
    joints=(crank, pin, fix1, fix2),
    order=(crank, pin, fix1, fix2),
    name="My four-bar linkage"
)

# Visualization
pl.show_linkage(my_linkage)
```
But I got the following error:

```
Traceback (most recent call last):
  File "LinkageOptimizer\linkage_opt.py", line 39, in <module>
    pl.show_linkage(my_linkage)
  File "Python\Python39\site-packages\pylinkage\visualizer.py", line 257, in show_linkage
    loci = tuple(tuple(i) for i in linkage.step(
  File "Python\Python39\site-packages\pylinkage\visualizer.py", line 257, in <genexpr>
    loci = tuple(tuple(i) for i in linkage.step(
  File "Python\Python39\site-packages\pylinkage\linkage.py", line 550, in step
    j.reload()
  File "Python\Python39\site-packages\pylinkage\linkage.py", line 187, in reload
    self.x, self.y = cyl_to_cart(self.r, self.angle + rot,
TypeError: unsupported operand type(s) for +: 'NoneType' and 'float'
```

**Comment by HugoFara** [comment]
Hi! Thanks for the drawings, it helped me a lot to understand what you wanted to do.

Basically, you didn't set the position of the `Fixed` joints, resulting in issues: position of `fix1` is `None`, so when trying to update it it was resulting in the wrong sum `'NoneType' and 'float'`.

For your first linkage (the box-shaped one), this changes should be enough (don't forget to import pi):
```python
# The triangle on the couple link
fix1 = pl.Fixed(joint0=crank, joint1=pin, distance=1.5, angle=pi/2)
fix2 = pl.Fixed(joint0=fix1, joint1=pin, distance=3, angle=pi/7)
```
This is a screenshot from the resulting animation:
![image](https://user-images.githubusercontent.com/72709893/192933806-97935e8b-0615-4e0d-b296-f4f5b3fdebda.png)

For the second linkage just removing `fix2` from the linkage and moving `fix1` should be enough. Hope it was what you needed!
